conjur/init/init_controller.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+
+"""
+InitController module
+
+This module is the controller that facilitates all init actions
+required to successfully configure the conjurrc
+"""
+
+# Builtins
+import logging
+import os
+import sys
+
+# Third party
+from urllib.parse import urlparse
+
+# Internals
+import conjur.constants
+
+class InitController:
+    """
+    InitController
+
+    This class represents the Presentation Layer for the init command
+    """
+    force_overwrite = False
+    conjurrc_data = None
+    init_logic = None
+
+    def __init__(self, conjurrc_data, init_logic, force):
+        self.conjurrc_data = conjurrc_data
+        self.init_logic = init_logic
+        self.force_overwrite = force
+
+    def load(self):
+        """
+        Method that facilitates all method calls in this class
+        """
+        fetched_certificate = self.get_server_certificate()
+        self.write_certificate(fetched_certificate)
+
+        self.get_account_info(self.conjurrc_data)
+
+        self.write_conjurrc()
+
+        sys.stdout.write("Configuration initialized successfully!\n")
+        sys.stdout.write("To begin using the CLI, log in to the Conjur server by" \
+                         "running `conjur authn login`")
+
+    def get_server_certificate(self):
+        """
+        Method to get the certificate from the Conjur endpoint detailed by the user
+        """
+        # pylint: disable=line-too-long
+        if self.conjurrc_data.appliance_url is None:
+            self.conjurrc_data.appliance_url = input("Enter the URL of your Conjur server: ")
+
+        url = urlparse(self.conjurrc_data.appliance_url)
+        if url.scheme == "http":
+            return None
+
+        if self.conjurrc_data.cert_file is not None:
```

Comment:
If cert file does not exist, what will be the error?
Do you want to check it, or you trust the error handling to do it?

Reply:
why does an error need to be returned? we take the cert the user provides for us and we don't need to fetch the cert

Reply:
no, you take the cert **path**, but if the file it points to does not exist, what will happen?